### DISCUSSION TITLE
Sonoff sensors drain their batteries - the reason

### DISCUSSION THREAD

**Discussion by petahl** [Koenkk/zigbee2mqtt#21236] [Devices]
I have a Sonoff ZBDongle-E as Coordinator for 2 SNZB-04 (Door sensor) and one SNZB-03 (Motion sensor). The sensors are only paired, no other actions taken.

All sensors report every 80 minutes with the sequence below as an example.
**Debug** 2024-02-05 11:06:45Received Zigbee message from 'Entrance', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryVoltage":30}' from endpoint 1 with groupID 0
**Info** 2024-02-05 11:06:45MQTT publish: topic 'zigbee2mqtt/Entrance', payload '{"battery":100,"battery_low":false,"contact":true,"linkquality":132,"tamper":false,"voltage":3000}'
**Debug** 2024-02-05 11:06:45Received Zigbee message from 'Entrance', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 1 with groupID 0
**Info** 2024-02-05 11:06:45MQTT publish: topic 'zigbee2mqtt/Entrance', payload '{"battery":100,"battery_low":false,"contact":true,"linkquality":124,"tamper":false,"voltage":3000}'
**Debug** 2024-02-05 11:11:24Saving state to file /opt/zigbee2mqtt/data/state.json

Passive units should only send Availability messages to show they are alive, in my opinion, and not very often!
Is the reason to update the state.json file?
The behaviour causes battery drain!
**How do I get rid of useless messages?**

I am still learning about Zigbee and try to use Zigbee sensors together with "homemade sensors" in a Surveillance system (python program). Probably I have missed some configuration detail?

**Comment by DubhAd** [comment]
> **How do I get rid of useless messages?**

Those are _sent_ by the sensor, so the answer is you can't.

They're also unlikely to be the source of battery drain. That's usually down to:
1. Weak signal
2. Interference
3. Bad hardware

**Reply by petahl** [reply]
LQI for the sensors are 132, 160 and 168 and they were quite easy to pair.
I did some googling and found evidence of that communication is the process taking most power in an IoT sensor. And you are correct that your 3 items on the list involve communication. Reducing need of communication thus should drain battery less.
I also have an Ikea Parasoll Door sensor and it is just doing it's job not chatter a lot of nonsense.
Unfortunately it looks like Bad hardware is the main reason. DO NOT BUY SONOFF SENSORS!
